peer/brontide.go

```diff
@@ -3617,6 +3590,20 @@ func (p *Brontide) StartTime() time.Time {
 // message is received from the remote peer. We'll use this message to advance
 // the chan closer state machine.
 func (p *Brontide) handleCloseMsg(msg *closeMsg) {
+	link := p.fetchLinkFromKeyAndCid(msg.cid)
+
+	if link != nil {
```

Comment:
No because the ChanCloser state machine needs to be driven. The Link's lifecycle terminates when shutdown has been both sent and received, but there are still messages that must be exchanged in the coop close negotiation after that.

Reply:
Makes sense, since after we mark the channel as borked, it won't ever be loaded into the link, so we can sorta fast forward past the flush state (or just accelerate by moving it to insta flushed).